### PR TITLE
 [Issue-3515] Notification service

### PR DESCRIPTION
**Pull request by bluezdot**

### CODE REVIEW

**Comment by saltict** [comment]
🚀 Deployed on https://pr-3687--sw-web-runner.netlify.app

File: packages/extension-koni-ui/src/Popup/Home/index.tsx [hunk 1]
```diff
@@ -125,7 +125,7 @@ function Component ({ className = '' }: Props): React.ReactElement<Props> {
           <Layout.Home
             onClickSearchIcon={onOpenGlobalSearchToken}
             showFilterIcon
-            showSearchIcon
+            showNotificationIcon
```
Comment (S2kael) [inline packages/extension-koni-ui/src/Popup/Home/index.tsx:128]: Remove search icon?

**Review by S2kael — CHANGES_REQUESTED** [review]

File: packages/extension-koni-ui/src/Popup/Home/index.tsx (hunk 1 above)
Comment (bluezdot) [reply packages/extension-koni-ui/src/Popup/Home/index.tsx:128]: request from BA, design

File: packages/extension-koni-ui/src/components/Layout/base/Home.tsx [hunk 2]
```diff
@@ -3,32 +3,52 @@
...
+
+    return isAllAccount ? Object.values(unreadNotificationCountMap).reduce((acc, val) => acc + val, 0) : unreadNotificationCountMap[currentAccountProxy.id] || 0;
+  }, [currentAccountProxy, isAllAccount, unreadNotificationCountMap]);
 
   const onOpenCustomizeModal = useCallback(() => {
     activeModal(CUSTOMIZE_MODAL);
   }, [activeModal]);
 
+  const onOpenNotification = useCallback(() => {
+    navigate('/settings/notification');
```
[the remark this answers is not in the document]
Comment (S2kael) [reply packages/extension-koni-ui/src/components/Layout/base/Home.tsx:49]: No change, confirmed with BA

File: packages/extension-koni-ui/src/components/Layout/base/Home.tsx [hunk 3]
```diff
@@ -60,15 +80,34 @@ const Home = ({ children, isDisableHeader, onClickFilterIcon, onClickSearchIcon,
       });
     }
 
+    if (showNotificationIcon) {
+      icons.push({
+        icon: (
+          <div className={'notification-icon'}>
```
[the remark this answers is not in the document]
Comment (S2kael) [reply packages/extension-koni-ui/src/components/Layout/base/Home.tsx:86]: Must restyle too much, not change component

File: packages/extension-base/src/services/inapp-notification-service/index.ts [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+  }
+
+  async validateAndWriteNotificationsToDB (notifications: _BaseNotificationInfo[], address: string) {
+    const proxyId = this.keyringService.context.belongUnifiedAccount(address) || address;
+    const accountName = this.keyringService.context.getCurrentAccountProxyName(proxyId);
+    const passNotifications: _NotificationInfo[] = [];
+
+    const [comparedNotifications, remindTimeConfig] = await Promise.all([
+      this.fetchNotificationsByParams({ notificationTab: NotificationTab.ALL, proxyId }),
+      await fetchLastestRemindNotificationTime()
```
Comment (S2kael) [inline packages/extension-base/src/services/inapp-notification-service/index.ts:147]: Fetching everytime?

**Review by S2kael — CHANGES_REQUESTED** [review]

**Pull request merged**